### ISSUE TITLE
CustomDropout Class

### ISSUE THREAD

**Issue by madhavajay** [DanyWind/fastai_bayesian_nn#2]
I am trying to understand the purpose of the CustomDropout module.
I see that the intention is to enable Dropout.

However in your example code you are using predict_with_mc_dropout:
```
def predict_entropy(img,n_times=10):
    pred = learn.predict_with_mc_dropout(img,n_times=n_times)
    probs = [prob[2].view((1,1) + prob[2].shape) for prob in pred]
    probs = torch.cat(probs)
    e = entropy(probs)
    return e
```

If you follow the code from here:
https://github.com/fastai/fastai/blob/b118a611729d32a45dd1293f416d6e743df109be/fastai/basic_train.py#L427

To here:
https://github.com/pytorch/pytorch/blob/0d7391f8b2cc221d330b56f3d8971a2d4ef9c741/torch/nn/modules/module.py#L1045

And then finally through to here:
https://github.com/pytorch/pytorch/blob/97c1e90f46230ee16605765d57202b60622511e5/torch/nn/functional.py#L788

You can see that predict_with_mc_dropout should be enabling "training" which in turn will be enabling dropout.

Is the CustomDropout module needed and if so what for?

**Comment by dhuynh95** [comment]
Hello there :) 

Well I could have coded it myself but was too lazy as I saw someone already did.

The advantage of using **predict_with_mc_dropout** are twofold : first I don't need to recode it, second it works on individual predictions.

CustomDropout main's goal is to have a different switch than the module.training to activate or not the stochasticity, because during inference, Fastai will turn your learn.model.training to False, therefore deactivating your Dropout if you use a regular one. You could turn it back on separetely but I just prefer to create my own CustomDropout with its own switch so that I never forget to turn it on.

I wanted to keep the API of Fastai with something similar to **get_preds**, which acts on datasets rather than single observations, that is why I coded **get_preds_with_sample**, but did not code one for single observations as it already existed.

**Comment by madhavajay** [comment]
@DanyWind okay so if I understand correctly, you wanted to use the more advanced interface get_preds which does not contain a with_dropout flag. But there is no difference in the final result so I can avoid using the CustomDropout class and just apply the metrics to the output of predict_with_mc_dropout or any predict method and with_dropout=True enabled?

I just wanted to understand if it was doing something special that mc_dropout was not doing.
